### ISSUE TITLE
[Enhancement]: Add a confirmation modal when the user tries to navigate away with unsaved changes

### ISSUE THREAD

**Issue by jarekmorawski** [woothemes/woocommerce#35194]
### Describe the solution you'd like

We should add a browser confirmation modal to prevent accidental data loss when the user navigates to a different page while filling out the product form. For example, if I add images, but don't save the product as a draft, I should see the modal when I try to go to a different page or close the tab.

**Acceptance criteria**

- [ ] The user sees a modal a confirmation modal when they try to leave the editor or close the tab without saving changes.
- [ ] The behavior is consistent with the Gutenberg editor.

### Additional context

<img width="372" alt="Screen Shot 2022-10-19 at 15 02 41" src="https://user-images.githubusercontent.com/79307566/196698373-ab22ed4a-08c3-4e07-8388-4e23956e9cfe.png">

**Comment by mdperez86** [comment]
> The behavior is consistent with the Gutenberg editor.

@mattsherman @octaedro do you know about this? Where I can find it? How to see it?

**Comment by jarekmorawski** [comment]
> Where I can find it? How to see it?

* On your test site, navigate to `Posts > Add New`
* Enter any text. For example, the title and some body content
* Don't save the draft
* Close the tab or click any menu item to navigate away from the page

Here's the video from my test site.

https://user-images.githubusercontent.com/79307566/197749006-d6ddddf0-6a07-4640-8569-c55e9da1e598.mov

**Comment by mdperez86** [comment]
@jarekmorawski Awesome!!! Thanks

**Comment by octaedro** [comment]
We have another one in the OBW

![Screen Shot 2022-10-25 at 13 56 14](https://user-images.githubusercontent.com/1314156/197836221-fe65cdd2-54d2-4be5-bc87-76e5ecb5520a.png)


@jarekmorawski The idea here is to use the one you mentioned, right?

**Comment by mdperez86** [comment]
Estimation: 2

**Comment by jarekmorawski** [comment]
Interesting! I was thinking about the default one, but since we've already used a custom modal in https://github.com/woocommerce/woocommerce/pull/34948, it'd make sense to use it here too. 👍

I'd only change the copy to something more relevant.

> **Save changes?**
This product has unsaved changes. If you go to a different page, they'll be lost. Do you want to save the changes before you go?
[ Discard ] [ Save ]

**Comment by mdperez86** [comment]
I was considering to add the custom modal but I was not able to find a performant way of do it. After some researches I found the right way of do this is by using the window beforeunload event which is described [here](https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event#event_handler_aliases). Unfortunately, this way is not flexible at all so the only thing we can do is change the confirmation dialog message.
Also I took my time to check how the wizzard implementation was made and it doesn't fit the AC described in this issue because they open the modal only when the user click on a different step which is totaly different to what we have here.

According to the prev and unless anyone else in the team has a better solution for this I purpose the following:
- Implement this issue as it was described initialy. Not using a custom modal as a confirmation dialog.
- Have a team discussion to see if there is another way of implementing this considering performance and a custom modal.

cc: @jarekmorawski @pmcpinto
